Automattic/wp-calypso
# Store: Add support for universal analytics for stores

Closes #17105

To test:
* First, comment out these lines to avoid a bazillion annoying warnings due to another component in the settings > traffic page

https://github.com/Automattic/wp-calypso/blob/b855b7a8a7e9f6a24e4fc87e8fe251cfd10603fd/client/boot/app.js#L5-L7

https://github.com/Automattic/wp-calypso/issues/18592#issuecomment-346142563

* First, let's make sure you have a fresh clean Jetpack installed and connected
* Deactivate and delete the Jetpack plugin if you are running it
* Install the Jetpack plugin from the .org repository and activate and connect it
* Install the Jetpack beta plugin from https://jetpack.com/download-jetpack-beta/ and activate it
* Note: If you have WooCommerce Services 1.9.0 installed, you will get a PHP Notice when you deactivate Jetpack: Use of undefined constant JETPACK__VERSION. This can be ignored - it will be fixed in the next version of WooCommerce services - https://github.com/Automattic/woocommerce-services/issues/1249
* Go to /wp-admin/admin.php?page=jetpack-beta and activate the RC `5.6-beta2` release
* Note the Jetpack version you get on wp-admin/plugins.php and make sure it is 5.6 or higher
* Install and activate the Jetpack Google Analytics Helper plugin https://github.com/automattic/jetpack-google-analytics-helper
* Install this branch and navigate to http://calypso.localhost:3000/settings/traffic/{DOMAIN}
* Enter a valid Google Analytics Tracking ID and hit Save Settings (leave all the other settings off for now)
* Verify you are told settings saved successfully
* Open https://{DOMAIN}/wp-admin/tools.php?page=jpgah
* Make sure the settings there match what you see at http://calypso.localhost:3000/settings/traffic/{DOMAIN}
* Open https://{DOMAIN}
* Make sure if you view the page source, you see your tracking ID there
* Modify a few other settings on http://calypso.localhost:3000/settings/traffic/{DOMAIN} and save them
* Refresh https://{DOMAIN}/wp-admin/tools.php?page=jpgah
* Make sure the settings there match what you see at http://calypso.localhost:3000/settings/traffic/{DOMAIN}
* Bonus: Use the testing instructions on https://github.com/Automattic/jetpack/pull/8218 as a guide to ensure that your Google Analytics dashboard gets the expected data.

Wut It Looks Like:

<img width="1045" alt="screen shot 2017-11-21 at 3 51 37 pm" src="https://user-images.githubusercontent.com/1595739/33103849-c04e9a06-ced9-11e7-9b0e-fa38535aeec9.png">

<img width="1044" alt="screen shot 2017-11-21 at 3 52 08 pm" src="https://user-images.githubusercontent.com/1595739/33103850-c066e962-ced9-11e7-9f09-b8e2b524dc2d.png">
